tests/test_context.py

```diff
@@ -166,3 +166,80 @@ def test_build_order_name_canonicalization(tmp_context):
         },
     ]
     assert expected == contents
+
+
+def test_processing_constraints_file(tmp_context):
```

Comment:
I would add more edge cases since we already found one that didn't work before

Reply:
Okay, I will do that!
Also, I am not able to understand why e2e tests are failing